web/react/components/mention_list.jsx

```diff
@@ -17,14 +17,39 @@ module.exports = React.createClass({
     displayName: "MentionList",
     componentDidMount: function() {
         PostStore.addMentionDataChangeListener(this._onChange);
-
         var self = this;
-        $('body').on('keypress.mentionlist', '#'+this.props.id,
+
+        $('body').on('keydown.mentionlist', '#'+this.props.id,
             function(e) {
-                if (!self.isEmpty() && self.state.mentionText != '-1' && e.which === 13) {
+                if (!self.isEmpty() && self.state.mentionText != '-1' && (e.which === 13 || e.which === 9)) {
+                    e.stopPropagation();
+                    e.preventDefault();
+                    self.addCurrentMention();
+                }
+                else if (!self.isEmpty() && self.state.mentionText != '-1' && (e.which === 38 || e.which === 40)) {
                     e.stopPropagation();
                     e.preventDefault();
-                    self.addFirstMention();
+
+                    var tempSelectedMention = -1;
+                    self.refs['mention' + self.state.selectedMention].deselect();
+                    if (e.which === 38) {    
+                        if (self.getSelection(self.state.selectedMention - 1))
+                            self.setState({ selectedMention: self.state.selectedMention - 1, selectedUsername: self.refs['mention' + (self.state.selectedMention - 1)].props.username });
+                        else {
+                            while (self.getSelection(++tempSelectedMention))
+                                ; //Need to find the top of the list
```

Comment:
No it's fine the way it is, I'm 0/5 as well I'm just not used to seeing it that way. When we have a standards guide we can decide how we want to do it